### PR TITLE
More fixes to the Prow config files

### PR DESCRIPTION
**Pull request by chizhg**
1. Clean up some Makefiles
2. Fix regex for `pull-test-infra-validate-prow-yaml`
3. Remove `post-knative-prow-cluster-config-updater` as it's not needed anymore after we have `make -C prow/ deploy`
4. Fixes some config paths

/cc @kvmware

### CODE REVIEW

File: prow/knative-autobump-config.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ extraFiles:
 prefixes:
   - name: "Prow"
     prefix: "gcr.io/k8s-prow/"
-    refConfigFile: "prow/cluster/deck_deployment.yaml"
+    refConfigFile: "config/prow/cluster/deck_deployment.yaml"
```
Comment (kvmware) [inline prow/knative-autobump-config.yaml:20]: There shouldn't be any config/prow folder?

Comment (chizhg) [reply prow/knative-autobump-config.yaml:20]: This is referring to the config file in the kubernetes/test-infra repo, for which the path is https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/deck_deployment.yaml

Comment (kvmware) [reply prow/knative-autobump-config.yaml:20]: Did I miss you put testgrid there I guess? However,  there is no cluster folder there. As intended

Comment (chizhg) [reply prow/knative-autobump-config.yaml:20]: Sorry not sure I understood your question...

The Prow team is using k8s Prow instance as a "staging" environment, where the config files are in https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster. Downstream Prows such as Knative Prow are can use the `autobump` tool to bump to the same version as k8s Prow, and it parses the up-to-date image version from https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/deck_deployment.yaml, which is currently gcr.io/k8s-prow/deck:`v20220211-9762cdc433`

Comment (kvmware) [reply prow/knative-autobump-config.yaml:20]: I missed that it was referring to k8s/test-infra my bad sorry

**Pull request merged**